### ISSUE TITLE
classNamePrefix and classNameSuffix seems to don't work with the gradle plugin

### ISSUE THREAD

**Issue by tschob** [joelittlejohn/jsonschema2pojo#328]
Hi, 

I tried to set the classNamePrefix and classNameSuffix in the build.gradle with these 2 combinations:
```
String classNameSuffix = 'Suffix'
classNameSuffix = 'Suffix'
```
But the generated classes don't contain the suffix.

Did I miss sth.? Otherwise there seems to be a bug.

**Comment by ben-manes** [comment]
You can inspect the plugin's configuration by running with `--info`. If its not set then its a bug.

**Comment by ben-manes** [comment]
Running the Java example with the suffix configured as,

```gradle
jsonSchema2Pojo {
  targetPackage = 'example'
  includeJsr303Annotations = true
  propertyWordDelimiters = ['_'] as char[]
  classNameSuffix = 'Suffix'
}
```

outputs the conguration
```bash
...
classNamePrefix =
classNameSuffix = Suffix
```

The generated sources are inconsistent. The files under `build/generated-sources/js2p/example` lack the suffix, but it generates new classes under `build/generated-sources/js2p/org/joda/time` with the suffix (whereas this is a provided dependency and should not be generated).

This appears to not be a plugin bug, but something within the library itself.

**Comment by joelittlejohn** [comment]
I think a pull request made in the last few months applied some refactoring
to reuse code relating to class naming in a couple of places. Unfortunately
I think this has caused some subtle regressions here and there.

We should be able to expand the integration tests in this area and expose
the problem.

**Comment by ddossot** [comment]
The regressions seem to have affected Maven too because it doesn't work with the Maven plugin version `0.4.16`.

**Comment by pwilder** [comment]
I'm seeing the same behavior as ddossot but I think I may know the problem area. The following snippet is taken from the latest org.jsonschema2pojo.rules.ObjectRule class code.

```
try {
            boolean usePolymorphicDeserialization = usesPolymorphicDeserialization(node);
            if (node.has("javaType")) {
                String fqn = substringBefore(node.get("javaType").asText(), "<");

                if (isPrimitive(fqn, _package.owner())) {
                    throw new ClassAlreadyExistsException(primitiveType(fqn, _package.owner()));
                }

                int index = fqn.lastIndexOf(".") + 1;
                if (index >= 0 && index < fqn.length()) {
 
                    //===============
                    //We can see here that class prefix/suffix is applied
                    //===============
                    fqn = fqn.substring(0, index) + ruleFactory.getGenerationConfig().getClassNamePrefix() + fqn.substring(index) + ruleFactory.getGenerationConfig().getClassNameSuffix();
                }

                try {
                    _package.owner().ref(Thread.currentThread().getContextClassLoader().loadClass(fqn));
                    JClass existingClass = TypeUtil.resolveType(_package, fqn + (node.get("javaType").asText().contains("<") ? "<" + substringAfter(node.get("javaType").asText(), "<") : ""));

                    throw new ClassAlreadyExistsException(existingClass);
                } catch (ClassNotFoundException e) {
                    if (usePolymorphicDeserialization) {
                        newType = _package.owner()._class(JMod.PUBLIC, fqn, ClassType.CLASS);
                    } else {
                        newType = _package.owner()._class(fqn);
                    }
                }
            } else {
                //====================
                //Prefix and suffix do not appear to be applied in this if block
                //====================
                if (usePolymorphicDeserialization) {
                    newType = _package._class(JMod.PUBLIC, getClassName(nodeName, _package), ClassType.CLASS);
                } else {
                    newType = _package._class(getClassName(nodeName, _package));
                }
            }
        } catch (JClassAlreadyExistsException e) {
            throw new ClassAlreadyExistsException(e.getExistingClass());
        }

        ruleFactory.getAnnotator().propertyInclusion(newType, node);

        return newType;

    }
``` 

I think this can be address with relatively minimal change:



```
private String getClassName(String nodeName, JPackage _package) {
        String className = ruleFactory.getNameHelper().replaceIllegalCharacters(capitalize(nodeName));
        String normalizedName = ruleFactory.getNameHelper().normalizeName(className);
        return makeUnique(normalizedName, _package);
    }
```

To:
```
private String getClassName(String nodeName, JPackage _package) {
        String classNamePrefix = ruleFactory.getGenerationConfig().getClassNamePrefix();
        String classNameSuffix = ruleFactory.getGenerationConfig().getClassNameSuffix();
        String fullNodeName = createFullNodeName(nodeName, classNamePrefix, classNameSuffix);
        String className = ruleFactory.getNameHelper().replaceIllegalCharacters(capitalize(fullNodeName));
        String normalizedName = ruleFactory.getNameHelper().normalizeName(className);
        return makeUnique(normalizedName, _package);
    }

    private String createFullNodeName(String nodeName, String classNamePrefix, String classNameSuffix) {
        String returnString = nodeName;
        if (classNamePrefix != null) {
            returnString = classNamePrefix + returnString;
        }

        if (classNameSuffix != null) {
            returnString = returnString + classNameSuffix;
        }

        return returnString;
    }
```

This change seems to pass all the existing ITs as well as a few new tests I added myself. I'm quite new to GitHub but if it was helpful I could try to setup a pull request. Alternatively I'd encourage one of the more knowledgeable developers to evaluate the code above and see if a code change is appropriate.

**Comment by joelittlejohn** [comment]
@pwilder thanks for looking into this. I could take the code you've pasted here, but I don't want to deprive you of your first GitHub pull request :) I'm sure this will be the first of many so why not give it a go!

The basic process is:

1. Fork this repository
2. Clone your fork
3. Create a new branch (`git checkout -b`) commit your change to this branch
4. Push your branch to GitHub
5. Create the pull request

After you have completed step 4, GitHub will be showing a bug green button to create the pull request when you view your fork. There's a detailed guide at https://help.github.com/articles/using-pull-requests.

Good luck!

**Comment by pwilder** [comment]
https://github.com/joelittlejohn/jsonschema2pojo/pull/465

Let me know if I did anything silly like try to merge in the wrong direction. Also, please be advised that my manual testing was limited so I'm trusting the existing test suite that I didn't break any existing behaviour. However, if there are manual tests you would like me to run prior to a merge please let me know.

**Comment by joelittlejohn** [comment]
Fixed be #465
